### PR TITLE
Uidevice add extensions

### PR DESCRIPTION
**Pull request by vilapuigvila**
## Checklist

- [ ] New Extension
- [x] New Test
- [ ] Changed more than one extension, but all changes are related
- [ ] Trivial change (doesn't require changelog)

### CODE REVIEW

File: Sources/UIDeviceExtensions.swift [hunk 1]
```diff
@@ -85,6 +85,18 @@ extension UIDevice {
         return UIDevice.current.userInterfaceIdiom == UIUserInterfaceIdiom.pad
     }
 
+    /// EZSE: Returns path of documents directory
+    class func documentsDirectoryPath() -> String {
```
Comment (Khalian) [inline Sources/UIDeviceExtensions.swift]: public class func.

File: Sources/UIDeviceExtensions.swift [hunk 2]
```diff
@@ -85,6 +85,18 @@ extension UIDevice {
...
+        let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true) as [String]
+        return paths[0]
+    }
+    
+    /// EZSE: Returns path of documents directory caches
+    class func cachesDirectoryPath() -> String {
```
Comment (Khalian) [inline Sources/UIDeviceExtensions.swift]: public class func

File: Sources/UIDeviceExtensions.swift (hunk 1 above)
Comment (Khalian) [inline Sources/UIDeviceExtensions.swift]: I dont understand how this is a UIDeviceExtension? There is no reference to self or any static references to UIDevice anywhere here.

Comment (vilapuigvila) [inline Sources/UIDeviceExtensions.swift]: @Khalian I agree, where do you thing would be the best place to put it? Maybe NSFileManager?

Comment (Khalian) [inline Sources/UIDeviceExtensions.swift]: Yea that sounds about right. I would wait for one of the collaborators to comment.

File: EZSwiftExtensionsTests/UIDeviceTests.swift [hunk 3]
```diff
@@ -7,10 +7,29 @@
...
+        super.setUp()
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+    }
+    
+    func testGetDocumentsDirectoryPath() {
+        
+        let docPath = UIDevice.documentsDirectoryPath()
```
Comment (Khalian) [inline EZSwiftExtensionsTests/UIDeviceTests.swift]: This test does not really convey what exactly documentsDirectoryPath is supposed to accomplish.

File: Sources/UIDeviceExtensions.swift (hunk 1 above)
Comment (lfarah) [inline Sources/UIDeviceExtensions.swift]: I agree. Should be FileManagerExtensions.

@goktugyil? @piv199?

Comment (goktugyil) [inline Sources/UIDeviceExtensions.swift]: Ok, you need me to create the file?

Comment (Khalian) [inline Sources/UIDeviceExtensions.swift]: No I dont think its needed. I added CGPoint extensions before. So this project does not stop anyone.

Comment (goktugyil) [inline Sources/UIDeviceExtensions.swift]: There is a little trick to make sure new files are referenced properly in normal project, example project and tests. People usually miss some part, I should write a checklist about it sometime

**Comment by vilapuigvila** [comment]
This pull request can be closed, I moved code to a new branch.

**Pull request closed without merging**